app/uk/gov/hmrc/transitmovements/repositories/MovementsRepository.scala

```diff
@@ -143,7 +147,8 @@ class MovementsRepositoryImpl @Inject() (
         Codecs.playFormatCodec(MongoFormats.movementIdFormat),
         Codecs.playFormatCodec(MongoFormats.mrnFormat),
         Codecs.playFormatCodec(MongoFormats.offsetDateTimeFormat),
-        Codecs.playFormatCodec(MongoFormats.eoriNumberFormat)
+        Codecs.playFormatCodec(MongoFormats.eoriNumberFormat),
+        Codecs.playFormatCodec(MongoFormats.lrnFormat)
       )
```

Comment:
Can we add the LRN as an index as we're going to scan it all the time? If possible, can we mark it as a unique index too? If so, would it be possible to avoid the separate check we do (more thinking if we're rejected due to a duplicate, can we detect that in code and send the right error message?)

Reply:
Okay, in that case just add the index but don't make it unique because we still need fast read access. We'll keep the selector you've written in to do the duplicate check.

Reply:
Yes agree, will add the index